streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java

```diff
@@ -379,40 +569,35 @@ private Topology setupTopologyWithoutIntermediateUserTopic() {
         return builder.build();
     }
 
-    private void cleanGlobal(final String intermediateUserTopic, final Properties sslConfig) throws Exception {
+    private void cleanGlobal(final Properties sslConfig,
+                             final String resetScenario,
+                             final String resetScenarioArg) throws Exception {
         // leaving --zookeeper arg here to ensure tool works if users add it
-        final String[] parameters;
-        if (intermediateUserTopic != null) {
-            parameters = new String[]{
-                "--application-id", APP_ID + testNo,
+        final List<String> parameterList = new ArrayList<>(
+            Arrays.asList("--application-id", APP_ID + testNo,
                 "--bootstrap-servers", bootstrapServers,
                 "--input-topics", INPUT_TOPIC,
-                "--intermediate-topics", INTERMEDIATE_USER_TOPIC,
-                "--zookeeper", "localhost:2181"
-            };
-        } else {
-            if (sslConfig != null) {
-                final File configFile = TestUtils.tempFile();
-                final BufferedWriter writer = new BufferedWriter(new FileWriter(configFile));
-                writer.write(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG + "=SSL\n");
-                writer.write(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG + "=" + sslConfig.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG) + "\n");
-                writer.write(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG + "=" + sslConfig.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG) + "\n");
-                writer.close();
-
-                parameters = new String[]{
-                    "--application-id", APP_ID + testNo,
-                    "--bootstrap-servers", bootstrapServers,
-                    "--input-topics", INPUT_TOPIC,
-                    "--config-file", configFile.getAbsolutePath()
-                };
-            } else {
-                parameters = new String[]{
-                    "--application-id", APP_ID + testNo,
-                    "--bootstrap-servers", bootstrapServers,
-                    "--input-topics", INPUT_TOPIC
-                };
-            }
+                "--intermediate-topics", INTERMEDIATE_USER_TOPIC));
```

Comment:
If we add the intermediate topic for all cases, I am wondering if this is correct.

I understand that the test is still passing and we just log an error, but should StreamsResetter not return a non-zero return value to indicate that something went wrong?

Reply:
That's a good point. Maybe we should still add a boolean parameter to indicate if we should include `--intermediate-topics", INTERMEDIATE_USER_TOPIC` into the config lists?

Reply:
Yes. And we should make sure that the test fails if we do it wrong. That the test passes atm, indicates an issue in the reset tool. It only returns non-zero exit code if there was an fatal exception and we crash. But we should also return non-zero if anything else went wrong. Atm, we catch, print/log and move on -- we should remember if any error occurred though and only return zero if no error occurred at all.

I would recommend to use different return codes for different errors and document it. Should be part of the public API IMHO.

And add a test to check the return codes :)

Reply:
Should this be a separate PR out of the scope of the KIP though?

Reply:
Absolutely. It's a bug and we can file a JIRA -- thus, the bug fix can also re-introduce the it-then-else for the intermediate topic here.

Reply:
@jeqo do you want to add back the boolean indicator whether intermediate topic exists and should be removed before we merge this PR?